File: front-end/LibraryApp-MDB-Pro-Basic/src/app/model/Reservation.ts
import {Book} from './Book';
import {User} from './User';
import {Client} from './Client';
import {ReservationState} from './ReservationState';

export class Reservation {
  id: number;
  bookList: Book[];
  user: User;
  client: Client;
  startDate: Date;
  endDate: Date;
  reservationState: ReservationState;
  processedDate: Date;

}

File: back-end/src/main/java/com/library/model/Reservation.java
package com.library.model;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
@Entity
@Table(name = "reservations")
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToMany
    @JoinTable(name = "reservation_book",
            joinColumns = @JoinColumn(name = "reservation_id"),
            inverseJoinColumns = @JoinColumn(name = "book_id"))
    private List<Book> bookList = new ArrayList<>();

    @ManyToOne
    private User user;

    @ManyToOne
    private Client client;

    private LocalDate startDate;

    private LocalDate endDate;

    private LocalDateTime processedDate;

    @ManyToOne
    private ReservationState reservationState;
}
